website/static/js/share/stats.js

```diff
@@ -92,6 +100,90 @@ function timeGraph (data) {
     });
 }
 
+Stats.sourcesAgg = function(){
+    var sourcesQuery = {'match_all':{}};
+    var sourcesAgg = {'sources': utils.termsFilter('field','_type')};
+    return {'query' : sourcesQuery, 'aggregations': sourcesAgg ,'filters' : {}}
+};
+
+Stats.sourcesByDatesAgg = function(){
+    var dateTemp = new Date(); //get current time
+    dateTemp.setMonth(dateTemp.getMonth() - 3);
+    var threeMonthsAgo = dateTemp.getTime();
+    var dateHistogramQuery = {'match_all':{}};
+    var dateHistogramAgg = {'sourcesByTimes': utils.termsFilter('field','_type')};
+    dateHistogramAgg.sourcesByTimes['aggregations'] = {'articlesOverTime' :
+        utils.dateHistogramFilter('providerUpdatedDateTime',threeMonthsAgo)};
+    return {'query' : dateHistogramQuery, 'aggregations': dateHistogramAgg ,'filters' : {}}
+};
+
+Stats.timeSinceEpochInMsToMMYY = function(timeSinceEpochInMs)
+{
+    var d = new Date(0);
+    d.setUTCSeconds(timeSinceEpochInMs/1000);
+    return d.getMonth().toString() + '/' + d.getFullYear().toString().substring(2);
+};
+
+Stats.shareDonutGraphParser = function(data)
+{
+    var chartData = {};
+    chartData.name = 'shareDonutGraph';
+    chartData.columns = [];
+    chartData.colors = {};
+    chartData.type = 'donut';
+
+    var providerCount = 0;
+    var hexColors = utils.generateColors(data.aggregations.sources.buckets.length);
+    var i = 0;
+    data.aggregations.sources.buckets.forEach(
+        function (bucket) {
+            chartData.columns.push([bucket.key, bucket.doc_count]);
+            providerCount = providerCount + (bucket.doc_count ? 1 : 0);
+            chartData.colors[bucket.key] = hexColors[i];
+            i = i + 1;
+        }
+    );
+    chartData.title = providerCount.toString() + ' Provider' + (providerCount !== 1 ? 's' : '');
+    $('.c3-chart-arcs-title').text(chartData.title); //dynamically update chart title
+    return chartData
+};
+
+Stats.shareTimeGraphParser = function(data)
+{
+    var chartData = {};
+    chartData.name = 'shareTimeGraph';
+    chartData.columns = [];
+    chartData.colors = {};
+    chartData.type = 'area-spline';
+    chartData.x = 'x';
+    chartData.groups = [];
+    var grouping = [];
+    grouping.push('x');
+    var hexColors = utils.generateColors(data.aggregations.sourcesByTimes.buckets.length);
+    var i = 0;
+    var datesCol = [];
+    data.aggregations.sourcesByTimes.buckets.forEach( //TODO @bdyetton what would be nice is a helper function to do this for any agg returned by elastic
```

Comment:
This aggregation should work on arbitrary datetime aggregations though, which is probably abstraction enough for now.